dubbo-spring-boot/dubbo-spring-boot-observability-starter/src/main/java/org/apache/dubbo/spring/boot/observability/autoconfigure/otel/OpenTelemetryAutoConfiguration.java

```diff
@@ -0,0 +1,257 @@
+@ConditionalOnDubboTracingEnable
+@ConditionalOnClass({OtelTracer.class, SdkTracerProvider.class, OpenTelemetry.class})
+@EnableConfigurationProperties(DubboTracingProperties.class)
+public class OpenTelemetryAutoConfiguration {
+
+    /**
+     * Default value for application name if {@code spring.application.name} is not set.
+     */
+    private static final String DEFAULT_APPLICATION_NAME = "application";
+
+    private final DubboTracingProperties dubboTracingProperties;
+
+    OpenTelemetryAutoConfiguration(DubboTracingProperties dubboTracingProperties) {
+        this.dubboTracingProperties = dubboTracingProperties;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    OpenTelemetry openTelemetry(SdkTracerProvider sdkTracerProvider, ContextPropagators contextPropagators) {
+        return OpenTelemetrySdk.builder().setTracerProvider(sdkTracerProvider).setPropagators(contextPropagators)
+            .build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    SdkTracerProvider otelSdkTracerProvider(Environment environment, ObjectProvider<SpanProcessor> spanProcessors,
+                                            Sampler sampler) {
+        String applicationName = environment.getProperty("spring.application.name", DEFAULT_APPLICATION_NAME);
+        SdkTracerProviderBuilder builder = SdkTracerProvider.builder().setSampler(sampler)
+            .setResource(Resource.create(Attributes.of(ResourceAttributes.SERVICE_NAME, applicationName)));
+        spanProcessors.orderedStream().forEach(builder::addSpanProcessor);
+        return builder.build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    ContextPropagators otelContextPropagators(ObjectProvider<TextMapPropagator> textMapPropagators) {
+        return ContextPropagators.create(TextMapPropagator.composite(textMapPropagators.orderedStream().collect(Collectors.toList())));
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    Sampler otelSampler() {
+        Sampler rootSampler = Sampler.traceIdRatioBased(this.dubboTracingProperties.getSampling().getProbability());
+        return Sampler.parentBased(rootSampler);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    SpanProcessor otelSpanProcessor(ObjectProvider<SpanExporter> spanExporters,
+                                    ObjectProvider<SpanExportingPredicate> spanExportingPredicates, ObjectProvider<SpanReporter> spanReporters,
+                                    ObjectProvider<SpanFilter> spanFilters) {
+        return BatchSpanProcessor.builder(new CompositeSpanExporter(spanExporters.orderedStream().collect(Collectors.toList()),
+            spanExportingPredicates.orderedStream().collect(Collectors.toList()), spanReporters.orderedStream().collect(Collectors.toList()),
+            spanFilters.orderedStream().collect(Collectors.toList()))).build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    Tracer otelTracer(OpenTelemetry openTelemetry) {
+        return openTelemetry.getTracer("org.apache.dubbo");
```

Comment:
Could we add the Dubbo version:
![image](https://user-images.githubusercontent.com/56248584/221387480-ada681b2-d1d7-490b-b931-dfd76f627a14.png)

Reply:
Yes